staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fq/fq.go

```diff
@@ -0,0 +1,468 @@
+package fq
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// FQScheduler is a fair queuing implementation designed for the kube-apiserver.
+// FQ is designed for
+// 1) dispatching requests to be served rather than packets to be transmitted
+// 2) serving multiple requests at once
+// 3) accounting for unknown and varying service times
+// implementation of:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
+type FQScheduler struct {
+	lock sync.Mutex
+	// PublicLock is used for complicated operations which span
+	PublicLock       sync.Mutex
+	Queues           []FQQueue
+	clock            clock.Clock
+	vt               float64
+	C                int
+	G                float64
```

Comment:
even tho `C` and `G` are denoted from the KEP, a complete name may help readablity, wdyt?

Reply:
updated.  C is now consolidated into 'concurrencyLimit'.  G is now 'estimatedServiceTime'